ConvertKit/ConvertKit-WordPress
# Improve performance of our javascript for fun and profit

Closes #127 

## Summary
Before this PR, our javascript makes an ajax request on _every page load_ 😱 

This PR adds a few things:
- An option to turn off loading of our javascript entirely. This is the Nuclear Option™ and will prevent tagging and custom content (using our shortcode, and in the future a Gutenberg block) from functioning.
- Only sends an ajax request to tag subscribers on page load _if_ we are on a page with tags AND we have a subscriber id (either via cookie or URL query param).

## QA
- Use this version of the plugin: https://github.com/ConvertKit/ConvertKit-WordPress/archive/issue/127.zip
- Set up a page or post with a tag assigned to it
- Set up a page or post without a tag assigned to it
- Visit both pages/posts without a `ck_subscriber_id` (e.g. in an incognito window) and verify that no requests are made to `admin-ajax.php`
- Visit a page with a tag assigned to it, with a `ck_subscriber_id` (e.g. by appending `?ck_subscriber_id={id_here}` to the URL) that belongs to the account the site is connected to, and verify that a request is made to `admin-ajax.php`
- Verify that the above page load also triggers the assigned tag being added to the subscriber in ConvertKit
- Visit a page without a tag assigned, but with a `ck_subscriber_id` (e.g. in the URL as above), and verify that no request is made to `admin-ajax.php`
- Check the box in the settings area to disable ConvertKit javascript entirely, then visit pages both with and without tags assigned and verify that `wp-convertkit.js` is not loaded.
- Re-enable ConvertKit javascript in the settings, then visit a page with a ConvertKit form on it, and subscribe through the form. Verify that a request is made to `admin-ajax.php` with `action: ck_get_subscriber` in the Form Data portion of the request.

## Ready for review?
- [x] Add "ready for review" label
- [x] Assign a reviewer (or two)
- [x] post RFR in #wordpress
